### PR TITLE
CI add codecov token to the config

### PR DESCRIPTION
**Pull request by adrinjalali**
The codecov token we have is not actually used in PRs since it's a token on GH, and they're not exposed to `pull_request` triggered actions.

The recommended action seems to be adding it hard coded, which doesn't really pose any security risks: https://github.com/codecov/codecov-action/issues/837#issuecomment-1453877750

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -5,5 +5,6 @@ codecov:
   notify:
     after_n_builds: 12
     wait_for_ci: true
+  token: 2b8d4d69-6de6-4e1d-840a-5ccf9d849565
```
Comment (BenjaminBossan) [inline .codecov.yml:8]: How about a comment about why it's okay to include the token here?

**Review by BenjaminBossan — COMMENTED** [review]
Ah I see, it's the same issue again about exposing secrets to forks. I wonder if GH could fix that without leaking the secret, but even if possible, we can't wait for it.

File: .codecov.yml (hunk 1 above)
Comment (adrinjalali) [reply .codecov.yml:11]: Added comment.

**Comment by adrinjalali** [comment]
I don't think it's something GH wants to "fix".

**Review by BenjaminBossan — APPROVED** [review]

**Comment by BenjaminBossan** [comment]
> I don't think it's something GH wants to "fix".

Yeah, at this point probably not.

**Pull request merged**